model/task/task.go

```diff
@@ -2572,35 +2580,90 @@ func ArchiveMany(tasks []Task) error {
 	defer session.EndSession(ctx)
 
 	txFunc := func(sessCtx mongo.SessionContext) (interface{}, error) {
-		oldTaskColl := evergreen.GetEnvironment().DB().Collection(OldCollection)
-		_, err = oldTaskColl.InsertMany(ctx, archived)
-		if err != nil {
-			return nil, err
+		var err error
+		if len(toArchive) > 0 {
+			oldTaskColl := evergreen.GetEnvironment().DB().Collection(OldCollection)
+			_, err = oldTaskColl.InsertMany(sessCtx, toArchive)
+			if err != nil {
+				return nil, errors.Wrap(err, "archiving tasks")
+			}
+		}
+		if len(taskIds) > 0 {
+			_, err = evergreen.GetEnvironment().DB().Collection(Collection).UpdateMany(sessCtx,
+				bson.D{{Key: "_id", Value: bson.D{{Key: "$in", Value: taskIds}}}}, // Query all display tasks and tasks
+				bson.A{ // Pipeline
+					bson.D{{Key: "$set", Value: bson.D{ // execution = execution + 1
+						{Key: ExecutionKey, Value: bson.D{
+							{Key: "$add", Value: bson.A{
+								"$" + ExecutionKey, 1,
+							}},
+						}},
+					}}},
+					bson.D{{Key: "$unset", Value: bson.A{
+						AbortedKey,
+						AbortInfoKey,
+						OverrideDependenciesKey,
+					}}}})
+			if err != nil {
+				return nil, errors.Wrap(err, "archiving tasks")
+			}
 		}
+		if len(execTasksIds) > 0 {
+			// Backwards compatibility call + LPE setting for all tasks
+			_, err = evergreen.GetEnvironment().DB().Collection(Collection).UpdateMany(sessCtx,
+				bson.D{{Key: "_id", Value: bson.D{{Key: "$in", Value: execTasksIds}}}}, // Query all execution tasks
+				bson.A{ // Pipeline
+					bson.D{{Key: "$set", Value: bson.D{ // Sets LatestParentExecution (LPE) = !exists(LPE) ? execution + 1 : LPE + 1
+						{Key: LatestParentExecutionKey, Value: bson.D{
```

Comment:
We shouldn't need to use this bson.D type / bson.A formatting for UpdateMany; the deleted code below is an example of this.

Reply:
Since old things can be reset we probably should keep the backwards compatible call though actually.

Reply:
I updated to use bson.M instead of bson.D, but for bson.A (which is bson representation of array), I think there isn't any alternative and looks nice (easier to see with the bson.M!). The alternative which we had before was a bson.M with key and value with value being "". Which seems more verbose. What do you think?

Reply:
I'm not sure I understand what you mean by there isn't an alternative. This also can just be bson.M, as visible in the old way this was written: https://github.com/evergreen-ci/evergreen/blob/c8ac8688404d6dd1ac0e420e206238ef5895b98e/model/task/task.go#L2512

Reply:
There isn't an alternative to bson.A as in an unordered representation of a bson array. bson.M is an unordered key value list (as bson.D is an ordered key value list). The unset can be changed to a bson.M but I think it wouldn't be as semantic and it would add the quotes-- but I think it has the benefit of matching what we originally did and matches more what we do in the codebase. I think the tradeoffs lean towards using bson.A to be more semantic but which do you recommend?

Reply:
I tried setting the pipeline $unset to bson.M and I get

write exception: write errors: [$unset specification must be a string or an array]

Reply:
But for the old UpdateMany call that you linked in the other thread, when using that, $unset MUST be bson.M

So it seems that:
- UpdateMany old call (not pipeline): $unset must be bson.M
- UpdateMany new call (uses pipeline): $unset must be bson.A

Reply:
So what I don't understand is why we're using pipeline at all. Why can't this just be the following? Is it strictly needed to use conditional? 
```
_, err = evergreen.GetEnvironment().DB().Collection(Collection).UpdateMany(sessCtx,
				bson.M{"_id": bson.M{"$in": execTaskIds}}, // Query all execution tasks
				bson.M{"$set": bson.M{ // Sets LatestParentExecution (LPE) = !exists(LPE) ? execution + 1 : LPE + 1
						LatestParentExecutionKey: bson.M{
							"$cond": bson.A{
								bson.M{"$not": bson.A{ // !exists(LPE)
									"$" + LatestParentExecutionKey,
								}},
								bson.M{"$add": bson.A{ // execution + 1
									"$" + ExecutionKey, 1,
								}},
								bson.M{"$add": bson.A{ // LPE + 1
									"$" + LatestParentExecutionKey, 1,
								}},
						},
					}},
				})
```

Reply:
Yeah, $cond is a pipeline restricted feature: https://www.mongodb.com/docs/v4.2/reference/operator/aggregation/cond/

Reply:
I tried without a pipeline and I get `The dollar ($) prefixed field '$cond' in 'latest_parent_execution.$cond' is not valid for storage.]` for additional info